apps/ex_nvr_web/lib/ex_nvr_web/live/dashboard_live.ex

```diff
@@ -263,6 +290,14 @@ defmodule ExNVRWeb.DashboardLive do
     end
   end
 
+  def handle_event("download_snapshot", %{"device" => device_id}, socket) do
+    socket
+    |> push_event("download-snapshot", %{
+      url: ~p"/api/devices/#{device_id}/snapshot"
```

Comment:
What's this ??
This will get you the current live snapshot from the rtsp stream which is not what I said in the issue. I want to get the current snapshot from the playable stream on the front-end. Means you need to `canvas` and the `hls.js` player for that. No back-end endpoint is involved.